src/query/algorithms/triangles/Triangles.cpp

```diff
@@ -86,108 +84,99 @@ long Triangles::run(JasmineGraphHashMapLocalStore graphDB, JasmineGraphHashMapCe
         degreeMap[degree].insert(startVId);
     }
 
     long triangleCount = 0;
-    long varOne = 0;
-    long varTwo = 0;
-    long varThree = 0;
-
-    std::map<long, std::map<long, std::vector<long>>> triangleTree;
-    std::vector<long> degreeListVisited;
-
-    std::map<long, std::set<long>>::iterator iterator;
+    std::unordered_map<long, std::unordered_map<long, std::unordered_set<long>>> triangleTree;
 
-    for (iterator = degreeMap.begin(); iterator != degreeMap.end(); ++iterator) {
-        long key = iterator->first;
-        std::set<long> vertices = iterator->second;
+    for (auto iterator = degreeMap.begin(); iterator != degreeMap.end(); ++iterator) {
+        std::set<long> &vertices = iterator->second;
 
-        if (key == 1) {
-            continue;
-        }
-        std::set<long>::iterator verticesIterator;
-
-        for (verticesIterator = vertices.begin(); verticesIterator != vertices.end(); ++verticesIterator) {
+        for (auto verticesIterator = vertices.begin(); verticesIterator != vertices.end(); ++verticesIterator) {
             long temp = *verticesIterator;
-            std::set<long> orderedUList(centralStore[temp].begin(), centralStore[temp].end());
-            std::set<long>::iterator uListIterator;
-            for (uListIterator = orderedUList.begin(); uListIterator != orderedUList.end(); ++uListIterator) {
-                long u = *uListIterator;
-                std::set<long> orderedNuList(centralStore[u].begin(), centralStore[u].end());
-                std::set<long>::iterator nuListIterator;
-                for (nuListIterator = orderedNuList.begin(); nuListIterator != orderedNuList.end(); ++nuListIterator) {
-                    long nu = *nuListIterator;
-                    if ( ((centralStore[temp].find(nu) != centralStore[temp].end()) ||
-                          (centralStore[nu].find(temp) != centralStore[nu].end())) &&
-                            (temp != nu) && (u != nu) && (temp != u) ) {
-                        std::vector<long> tempVector;
-                        tempVector.push_back(temp);
-                        tempVector.push_back(u);
-                        tempVector.push_back(nu);
-                        std::sort(tempVector.begin(), tempVector.end());
-
-                        varOne = tempVector[0];
-                        varTwo = tempVector[1];
-                        varThree = tempVector[2];
-
-                        std::map<long, std::vector<long>> itemRes = triangleTree[varOne];
-
-                        std::map<long, std::vector<long>>::iterator itemResIterator = itemRes.find(varTwo);
+            std::unordered_set<long> &unorderedUSet = centralStore[temp];
+            for (auto uSetIterator = unorderedUSet.begin(); uSetIterator != unorderedUSet.end(); ++uSetIterator) {
+                long u = *uSetIterator;
+                if (temp == u) continue;
+                std::unordered_set<long> &unorderedNuSet = centralStore[u];
```

Comment:
Instead of `Nu` in the `unorderedNuSet` can we have meaningful word there?

Reply:
Based on the offline discussion let leave it as it is. Here `Nu` means neighbor of U.